src/app/components/elements/panels/EventGenderDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { EventBookingDTO, UserRole } from "../../../../IsaacApiTypes";
+import { Col, Table, UncontrolledTooltip } from "reactstrap";
+import { asPercentage } from "../../../services";
+
+export const countEventDetailsByRole = (role: UserRole | undefined, eventBookings: EventBookingDTO[]) => {
+  const countByRole = {
+    genders: {
+      male: 0,
+      female: 0,
+      other: 0,
+      preferNotToSay: 0,
+      unknown: 0,
+    },
+    numberOfConfirmedOrAttendedBookings: 0,
+  };
+
+  eventBookings.forEach((booking) => {
+    const gender = booking.userBooked?.gender;
+    const bookingStatus = booking.bookingStatus;
+    const userRole = booking.userBooked?.role;
+
+    const roleFilter = role === undefined || userRole === role;
+
+    if (booking.userBooked && bookingStatus && ["CONFIRMED", "ATTENDED"].includes(bookingStatus) && roleFilter) {
```

Comment:
I think that the `&& bookingStatus` check could be superfluous here, since you're immediately checking if `bookingStatus` is confirmed or attended.

Reply:
`bookingStatus` can be undefined, so `&& bookingStatus` is required unless I wanted to do `.includes(bookingStatus!)`